packages/tanstack-react-query/src/internals/createOptionsProxy.ts

```diff
@@ -33,10 +33,30 @@ import {
   trpcSubscriptionOptions,
   type TRPCSubscriptionOptions,
 } from './subscriptionOptions';
-import type { QueryType, ResolverDef } from './types';
+import type { QueryType, ResolverDef, TRPCQueryKey } from './types';
 import { getQueryKeyInternal } from './utils';
 
-export interface DecorateQueryProcedure<TDef extends ResolverDef> {
+interface QueryKeyStuff {
+  /**
+   * Useful for query invalidation
+   *
+   * @see https://tanstack.com/query/latest/docs/framework/react/guides/query-invalidation
+   */
+  getQueryKey: () => TRPCQueryKey;
+
+  /**
+   * Simple utility to invalidate a query or router
+   *
+   * @see https://tanstack.com/query/latest/docs/framework/react/guides/query-invalidation
+   */
+  invalidate: () => Promise<void>;
+}
+
+export interface DecorateQueryProcedure<TDef extends ResolverDef>
+  extends Omit<QueryKeyStuff, 'getQueryKey'> {
```

Comment:
`invalidate` here will need to accept input right? so this extend isn't totally accurate.

Reply:
I'm not that happy with this solution by the way, it's just we need the methods in 3 places so I was trying to dry it out